api/configuration/v1alpha1/key_set_ref.go

```diff
@@ -19,13 +19,13 @@ const (
 // KeySetRef is the schema for the KeySetRef type.
 // It is used to reference a KeySet entity.
 // +kubebuilder:validation:XValidation:rule="self.type == 'namespacedRef' ? has(self.namespacedRef) : true", message="when type is namespacedRef, namespacedRef must be set"
+// +kubebuilder:validation:XValidation:rule="self.type == 'konnectID' ? has(self.konnectID) : true", message="when type is konnectID, konnectID must be set"
 // +apireference:kgo:include
 // +kong:channels=gateway-operator
 type KeySetRef struct {
 	// Type defines type of the KeySet object reference. It can be one of:
+	// - konnectID
 	// - namespacedRef
-	//
-	// +kubebuilder:validation:Enum=namespacedRef
```

Comment:
I believe we could use this enum validation but including the `konnectID`, WDYT?

Reply:
This is already validated on the `KeySetRefType` level.